src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClient.cs

```diff
@@ -440,6 +501,74 @@ private void PublishContactPoints()
 
             _contactPathsPublished = _contactPaths;
         }
+
+        private void ProcessSubscribeMessage(Subscribe subscribe, IActorRef receptionist)
+        {
+            if (RegisterExternalSubscriber(subscribe))
+                receptionist.Tell(new PublishSubscribe.Subscribe(subscribe.Topic, Self, subscribe.Group));
+        }
+
+        private void ProcessUnsubscribeMessage(Unsubscribe unsubscribe, IActorRef receptionist)
+        {
+            if (UnregisterExternalSubscriber(unsubscribe))
+                receptionist.Tell(new PublishSubscribe.Unsubscribe(unsubscribe.Subscribe.Topic, Self, unsubscribe.Subscribe.Group));
+        }
+
+        private bool RegisterExternalSubscriber(Subscribe subscribe)
```

Comment:
It looks like you're trying to add every possible subscriber to the client instead of using mediators:

1. This means, that if we have 10 subscribers on node A and 10 on node B, we need to send 20 messages remotelly from the client, while we could just send 2 (to mediators on each of the nodes) and keep the rest of messages send local only.
2. You're not watching over actor refs, which means, that if an actor dies, it will never get removed from the subscribers set. Ultimatelly you may end up with garbage actor refs, that are never removed, pilling up in memory + a lot of dead letter messages.